### PR TITLE
Rework concurrency in the signaling logic.

### PR DESCRIPTION
**Pull request by charlesmchen**
Adds an error, fixes a typo.

PTAL @michaelkirk

### CODE REVIEW

File: src/Messages/OWSMessageSender.m [hunk 1]
```diff
@@ -436,7 +436,7 @@ - (void)sendMessage:(TSOutgoingMessage *)message
         }
 
         if ([exception.name isEqualToString:OWSMessageSenderRateLimitedException]) {
-            NSError *error = OWSErrorWithCodeDescription(OWSErrorCodeSingalServiceRateLimited,
```
Comment (michaelkirk) [inline src/Messages/OWSMessageSender.m:439]: gets me every singal time.

File: src/Util/OWSError.m [hunk 2]
```diff
@@ -35,7 +37,13 @@
...
 {
     return OWSErrorWithCodeDescription(OWSErrorCodeFailedToSendOutgoingMessage,
-        NSLocalizedString(@"ERROR_DESCRIPTION_UNKNOWN_ERROR", @"Worst case generic error message"));
+                                       NSLocalizedString(@"ERROR_DESCRIPTION_UNKNOWN_ERROR", @"Worst case generic error message"));
+}
+
+NSError *OWSErrorMakeWebRTCMissingDataChannelError()
+{
+    return OWSErrorWithCodeDescription(OWSErrorCodeWebRTCMissingDataChannel,
+                                       NSLocalizedString(@"ERROR_DESCRIPTION_WEBRTC_MISSING_DATA_CHANNEL", @"Missing data channel while trying to sending message"));
```
Comment (michaelkirk) [inline src/Util/OWSError.m]: Is there any point in localizing an error like this? I don't think it would mean anything to our users or our translators.

**Review by michaelkirk — APPROVED** [review]
LGTM

File: src/Messages/OWSMessageSender.m (hunk 1 above)
Comment (charlesmchen) [reply src/Messages/OWSMessageSender.m:439]: =0

File: src/Util/OWSError.m (hunk 2 above)
Comment (charlesmchen) [reply src/Util/OWSError.m]: Yeah, I was wondering the same thing.

**Pull request merged**